org.jacoco.core/src/org/jacoco/core/internal/analysis/filter/SyntheticFilter.java

```diff
@@ -29,10 +30,19 @@ public void filter(final MethodNode methodNode,
 			return;
 		}
 
-		if (KotlinDefaultArgumentsFilter
-				.isDefaultArgumentsMethodName(methodNode.name)
-				&& KotlinGeneratedFilter.isKotlinClass(context)) {
-			return;
+		if (KotlinGeneratedFilter.isKotlinClass(context)) {
+			if (KotlinDefaultArgumentsFilter
+					.isDefaultArgumentsMethodName(methodNode.name)) {
+				return;
+			}
+
+			final Type methodType = Type.getMethodType(methodNode.desc);
```

Comment:
@Godin Move this code block to a static method in `KotlinCoroutinesFilter`?

Reply:
@marchof done